Main Readme (#69)
AnonymusRaccoon/Aeris: README.md

# Professional, personnal action-reaction manager

## What is Aeris

Aeris is an Action-Reaction system manager. It lets its users create *Pipelines*.

Pipelines are triggered by an *Action* (for example, when a PR is open on some GitHub repository), and perfoms *Reactions* (for example, play a given song on Spotify).

To manage your Pipeline, you can use the Android App, as well as the Web client.

Aeris supports a variety of service, on which Pipelines are perfomed upon:

- GitHub
- Spotify
- Youtube
- Discord

> - Gmail

Reviewer: It's not gmail it's anilist now.

- Twitter

> - Even more services in the future :smile:

Reviewer: Please delete that.

Reviewer: I don't think so.

## What do I need to use Aeris?

Make sure the following softwares are installed on your machine:

- Docker, (and Docker-Compose)

To setup Aeris, you need to provide the building system some information:

> - Client ID and Client Secrets for EACH services (fill the `.env.example`provided at the root of the repository)

Reviewer 2: ```suggestion
- Client ID and Client Secrets for EACH services (fill the `.env.example`provided at the root of the repository and rename it `.env`)
```

- A Host name, to let clients know how to call the API (usually localhost:8080)

## How to install it ?

To install Aeris, run the following commands at the root of the repository:

- `docker-compose -f docker-compose.yml build` (grab a snack, it might take some time)
- `docker-compose -f docker-compose.yml up`

## How to use it ?

> The Aeris server is accessible on the host's 8080 port

Reviewer 2: ```suggestion
The Aeris server is accessible on the host's 81 port
```

> You can access the Web client through port 8081

Reviewer 2: ```suggestion
You can access the Web client through `localhost`
```

Author: This is not compliant to the subject: https://intra.epitech.eu/module/2021/B-YEP-500/NAN-5-1/acti-480798/project/file/B-YEP-500_area.pdf

> An Android APK can be downloaded via localhost:8081/client.apk

Reviewer 2: ```suggestion
An Android APK can be downloaded via localhost/client.apk
```
